### PR TITLE
Changes to remove extraneous HandledPromise shim dependencies

### LINKED ISSUE #1469: Turn off eslint's `import/prefer-default-export` globally
We should turn off `import/prefer-default-export` globally. I thought we had. I've certainly written a lot of named exports without needing to suppress any warnings. I prefer named exports.

_Originally posted by @erights in https://github.com/_render_node/MDExOlB1bGxSZXF1ZXN0NDY3NTg0MDcw/pull_requests/unread_timeline_

### PR DESCRIPTION
**Pull request by michaelfig**
Closes #1469

We import the HandledPromise shim once-and-for-all in `@agoric/install-ses`.  Then, using a version of SES with Agoric/SES-shim#416, it is propagated as a global to all child compartments.

Removing the extraneous HandledPromise plumbing from the code will make it easier to migrate to the new `Promise.delegate` API once `@agoric/eventual-send/shim` implements it.

### CODE REVIEW

**Comment by warner** [comment]
I'm surprised the tests pass if all the Compartments are now missing their `HandledPromise` endowment entirely. Does that work because `install-ses` is now changing the global `Promise` to really be `HandledPromise`?

Take a second look at `install-metering-and-ses` and check that the ordering of (modify `globalThis.Promise`) and (instrument `globalThis`) doesn't leave us with a non-instrumented Promise.. I suspect there might be a problem there but I haven't dived in yet.

I also noticed this downgrades the `harden(E)` to `Object.freeze(E)`. Does that open up a communication channel via something like `E.when.message = 'hello'` ?

**Comment by michaelfig** [comment]
I'm not ready for review yet.

**Comment by michaelfig** [comment]
> I'm surprised the tests pass if all the Compartments are now missing their `HandledPromise` endowment entirely. Does that work because `install-ses` is now changing the global `Promise` to really be `HandledPromise`?

It will work with Agoric/SES-shim#416

> Take a second look at `install-metering-and-ses` and check that the ordering of (modify `globalThis.Promise`) and (instrument `globalThis`) doesn't leave us with a non-instrumented Promise.. I suspect there might be a problem there but I haven't dived in yet.

I've fixed this by ensuring the shim is installed before metering is turned on.

> I also noticed this downgrades the `harden(E)` to `Object.freeze(E)`. Does that open up a communication channel via something like `E.when.message = 'hello'` ?

I was hoping to split the restructuring changes (needed to disentangle harden from the vetted shim before lockdown) to eventual-send to a different PR, but I could do them in the current PR if that would be better.

**Comment by michaelfig** [comment]
@erights, this is of lower priority than Zoe RC.  I only expect a detailed review from you on the changes to `@agoric/eventual-send` that were necessary in order to make it a vetted shim.  Notably, we couldn't early-bind some global properties since the metering transform replaced them in the course of wrapping all the globals (including HandledPromise).

File: packages/promise-kit/src/promiseKit.js [hunk 1]
```diff
@@ -42,10 +40,12 @@ export function makePromiseKit() {
   /** @type {(reason: any) => void} */
   let rej = NOOP_INITIALIZER;
 
-  const p = new HandledPromise((resolve, reject) => {
-    res = resolve;
-    rej = reject;
-  });
+  const p =
+    /** @type {Promise<T> & { domain?: any  }} */
+    (new Promise((resolve, reject) => {
```
Comment (erights) [inline packages/promise-kit/src/promiseKit.js]: Are you sure? The more of the promises running around our system are handled promises, the more often we succeed at pipelining. Unless there's a reason not to, I prefer that promiseKit makes handled promises.

File: packages/produce-promise/src/producePromise.js [hunk 2]
```diff
@@ -4,8 +4,6 @@
 // eslint-disable-next-line spaced-comment
```
Comment (erights) [inline packages/produce-promise/src/producePromise.js:4]: Maybe kill this whole module first, before we invest any more effort maintaining it?

File: packages/promise-kit/src/promiseKit.js [hunk 3]
```diff
@@ -75,6 +75,6 @@ harden(makePromiseKit);
  * @returns {maybePromise is Promise} Whether it is a promise
  */
 export function isPromise(maybePromise) {
-  return HandledPromise.resolve(maybePromise) === maybePromise;
+  return Promise.resolve(maybePromise) === maybePromise;
```
Comment (erights) [inline packages/promise-kit/src/promiseKit.js:78]: Good point. isPromise is no longer precisely what HandledPromise.resolve tests.

**Review by erights — APPROVED** [review]
LGTM

File: packages/dapp-svelte-wallet/package.json [hunk 4]
```diff
@@ -21,10 +21,10 @@
...
   },
   "scripts": {
-    "lint-fix": "cd api&&yarn lint-fix",
-    "lint-check": "cd api&&yarn lint-check",
-    "test": "cd api&&yarn test",
-    "build": "cd ui&&yarn build"
+    "lint-fix": "exit 0",
+    "lint-check": "exit 0",
+    "test": "exit 0",
+    "build": "exit 0"
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/dapp-svelte-wallet/package.json:27]: The `ui` and `api` directories are already referenced in `agoric-sdk/package.json`, so we don't need to recurse to them manually.

File: packages/eventual-send/src/E.js [hunk 5]
```diff
@@ -1,5 +1,9 @@
 /* global harden */
 
+// FIXME: Remove this and just use harden once the E proxy no longer
+// is created by ../shim.js.
+const FIXMEharden = x => x;
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/eventual-send/src/E.js]: Ah, this is a good catch.  I will change this just to `harden`.

It used to be that `E` was constructed when the shim was loaded (which with a vetted shim, is before `harden` is available).  That's no longer the case.

File: packages/promise-kit/src/promiseKit.js (hunk 1 above)
Comment (michaelfig) [reply packages/promise-kit/src/promiseKit.js]: Okay, duly noted.  I changed this while debugging, but I will change it back to something like:

```js
const BetterPromise = globalThis.HandledPromise || Promise;
[...]
   new BetterPromise(...)
```

File: packages/eventual-send/src/no-shim.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+/* global HandledPromise */
+import makeE from './E';
+
+// eslint-disable-next-line import/prefer-default-export
+export const E = makeE(HandledPromise);
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/eventual-send/src/no-shim.js:4]: Really, only:
```js
import '@agoric/eventual-send/shim'; // Ensure the global HandledPromise exists
import { E } from '@agoric/eventual-send'; // User code: use the E proxy maker
```

I'll add this to the README.md.

File: packages/eventual-send/src/index.js [hunk 7]
```diff
@@ -45,7 +24,8 @@ export const E = makeE(hp);
  *
  * @return {typeof HandledPromise} Handled promise
  */
-export function makeHandledPromise(Promise) {
+// eslint-disable-next-line import/prefer-default-export
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/eventual-send/src/index.js]: It might be a peculiarity of my environment.  Will investigate.

File: packages/produce-promise/src/producePromise.js (hunk 2 above)
Comment (michaelfig) [reply packages/produce-promise/src/producePromise.js:4]: New issue #1471

File: packages/eventual-send/shim.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+/* global globalThis */
+
+// eslint-disable-next-line import/prefer-default-export
+import { makeHandledPromise } from './src/index';
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/eventual-send/shim.js:3]: New issue #1469

File: packages/eventual-send/src/index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/eventual-send/src/index.js]: We were missing some eslint rules.  Now fixed.

**Comment by michaelfig** [comment]
@FUDCo or @warner, would you be able to review the other changes (not `packages/eventual-send`)?  The criterion is whether this is indeed a good cleanup.

**Review by FUDCo — APPROVED** [review]
Three cheers for taking out the trash!

File: packages/install-ses/install-ses.js [hunk 9]
```diff
@@ -1,11 +1,12 @@
...
 import 'ses';
 
-// we need to enable Math.random as a workaround for 'brace-expansion' module
-// (dep chain: cosmic-swingset/ag-solo->temp->glob->minimatch->brace-expansion)
-// we need Date.now to build a timer device for cosmic-swingset
-lockdown({ mathTaming: 'unsafe', dateTaming: 'unsafe', errorTaming: 'unsafe' });
+// Install our HandledPromise global.
+import '@agoric/eventual-send/shim';
+
+lockdown({ errorTaming: 'unsafe' });
```
Comment (warner) [inline packages/install-ses/install-ses.js:9]: @michaelfig I'm surprised this works (and it seems unrelated to anything involving `HandledPromise`), what changed to allow either untaming to be removed?

Comment (michaelfig) [reply packages/install-ses/install-ses.js:9]: The modern SES start compartment now contains the original `Math.random` and `Date.now`.

Comment (warner) [reply packages/install-ses/install-ses.js:9]: oh, awesome, I missed that. That's the perfect solution.
